cli/cmd/deploy/deploy.go

```diff
@@ -358,6 +381,192 @@ func loginWithTelemetry(ctx context.Context, ch *cmdutil.Helper, redirectURL str
 	return nil
 }
 
+func createGithubRepoFlow(ctx context.Context, ch *cmdutil.Helper, localGitPath string, silent bool) error {
+	// Get the admin client
+	c, err := ch.Client()
+	if err != nil {
+		return err
+	}
+
+	res, err := c.GetGithubUserStatus(ctx, &adminv1.GetGithubUserStatusRequest{})
+	if err != nil {
+		return err
+	}
+
+	if !res.HasAccess {
+		ch.Telemetry(ctx).RecordBehavioralLegacy(activity.BehavioralEventGithubConnectedStart)
+
+		if res.GrantAccessUrl != "" {
+			// Print instructions to grant access
+			time.Sleep(3 * time.Second)
+			ch.Print("Open this URL in your browser to grant Rill access to Github:\n\n")
+			ch.Print("\t" + res.GrantAccessUrl + "\n\n")
+
+			// Open browser if possible
+			if !silent {
+				_ = browser.Open(res.GrantAccessUrl)
+			}
+		}
+	}
+
+	// Poll for permission granted
+	pollCtx, cancel := context.WithTimeout(ctx, pollTimeout)
+	defer cancel()
+	for {
+		select {
+		case <-pollCtx.Done():
+			return pollCtx.Err()
+		case <-time.After(pollInterval):
+			// Ready to check again.
+		}
+
+		// Poll for access to the Github's user account
+		pollRes, err := c.GetGithubUserStatus(ctx, &adminv1.GetGithubUserStatusRequest{})
+		if err != nil {
+			return err
+		}
+		if !pollRes.HasAccess {
+			// Sleep and poll again
+			continue
+		}
+
+		// Emit success telemetry
+		ch.Telemetry(ctx).RecordBehavioralLegacy(activity.BehavioralEventGithubConnectedSuccess)
+
+		ch.PrintfBold(`No git remote was found.
+
+Rill projects deploy continuously when you push changes to Github.
+Therefore, your project must be on Github before you deploy it to Rill.
+		`)
+
+		ch.Print(`			
+You can continue here and Rill can create a Github Repository for you or
+you can exit the command and create a repository manually.
+		
+		`)
+		if !cmdutil.ConfirmPrompt("Do you want to continue?", "", true) {
+			ch.PrintfBold(githubSetupMsg)
+			return nil
+		}
+
+		repoOwner := pollRes.Account
+		if len(pollRes.Organizations) > 0 {
+			repoOwners := []string{pollRes.Account}
+			repoOwners = append(repoOwners, pollRes.Organizations...)
+			ch.Print("\nYou also have access to organization(s)\n\n")
+			repoOwner = cmdutil.SelectPrompt("Select Github account", repoOwners, pollRes.Account)
+		}
+		// create and verify
+		githubRepository, err := createGithubRepository(ctx, ch, pollRes, localGitPath, repoOwner)
+		if err != nil {
+			return err
+		}
+
+		printer.ColorGreenBold.Printf("\nRepository %q created successfully\n\n", *githubRepository.Name)
```

Comment:
Can we print the URL instead? I think that would be more useful for people. Like `Successfully created repository on "https://github.com/account/project"`